lib/bundler/cli/add.rb

```
@@ -11,7 +11,14 @@ def initialize(options, gem_name)
 
     def run
       version = @options[:version].nil? ? nil : @options[:version].split(",").map(&:strip)
-      dependency = Bundler::Dependency.new(@gem_name, version, @options)
+
+      begin
+        dependency = Bundler::Dependency.new(@gem_name, version, @options)
+      rescue Gem::Requirement::BadRequirementError => e
+        Bundler.ui.error(e.message)
+        return
```

this should raise `InvalidOption` and give more context about what's invalid